outlook/src/taskpane/components/Company/CompanySection/CompanySection.tsx

```diff
@@ -106,50 +107,64 @@ class CompanySection extends React.Component<CompanySectionProps, CompanySection
 
     }
 
-    private getCompanyLayout= () => {
-        const company = this.state.company as Company;
-
-        if (company.isEmpty())
+    private updateCompanyRequest = () => {
+        if (!this.props.partner.email)
         {
-            let enrichAndCreate = null;
-            if (this.props.partner.isAddedToDatabase() && !this.enriched)
+            let enrichmentInfo = new EnrichmentInfo(EnrichmentInfoType.EnrichContactWithNoEmail,
+                _t("This contact has no email address, no company could be enriched."));
+            this.context.showTopBarMessage(enrichmentInfo);
+            return;
+        }
+
+        this.setState({isLoading: true});
+
+
+        const enrichRequest = sendHttpRequest(HttpVerb.POST, api.baseURL + api.enrichAndUpdateCompany,
+            ContentType.Json, this.context.getConnectionToken(), {
+                partner_id: this.props.partner.id,
+            }, true);
+
+        this.context.addRequestCanceller(enrichRequest.cancel);
+
+        enrichRequest.promise.then((response) => {
+            const parsed = JSON.parse(response);
+
+            if (parsed.result['enrichment_info'])
             {
-                enrichAndCreate = (
-                    <div>
-                        <div>
-                            {_t("No company attached to this contact")}
-                        </div>
-                        <div className="odoo-secondary-button" style={{margin: "8px auto auto auto"}} onClick={this.enrichCompanyForPartnerRequest}>
-                            {_t("Create a Company")}
-                        </div>
-                    </div>
-                );
+                const enrichmentInfo = new EnrichmentInfo(parsed.result['enrichment_info'].type,
+                    parsed.result['enrichment_info'].info);
+                this.context.showTopBarMessage(enrichmentInfo);
+
+                if (enrichmentInfo.type == EnrichmentInfoType.NoData)
```

Comment:
Best practice: triple equal operator.
I think your linter can be configured to tell you that.

Reply:
I think that for JavaScript that would surely be the case.

However for TypeScript the "==" will check if operands are of the same type and if it's not the case, a compilation error happens, for this reason I don't think that it's necessary to use "===" here.

From what I can see, in typescript "===" is usually used when one of the operands or both is of type "any", which is not the case here.

Reply:
Yes my bad, I get confused because they look so similar otherwise.